script/gen/cxx14/all

```diff
@@ -0,0 +1,419 @@
+                    else:
+                        raise RuntimeError("unexpected option type: %s" % otype)
+                    filep.write("    TaskInfo &set_%s_option(%s) noexcept {\n" % (
+                                option_name_to_snake_case(oname), otype))
+                    filep.write("        options[\"%s\"] = v;\n" % oname)
+                    filep.write("        return *this;\n")
+                    filep.write("    }\n")
+                    filep.write("\n")
+            omit_nl = True
+        elif isinstance(value, STR):
+            filep.write("    std::string %s;\n" % name)
+            filep.write("\n")
+            filep.write("    /// Allows to set '%s' using a fluent style\n" % name)
+            filep.write("    TaskInfo &set_%s(const std::string &s) noexcept {\n" % name)
+            filep.write("        %s = s;\n" % name)
+            filep.write("        return *this;\n")
+            filep.write("    }\n")
+        elif isinstance(value, float):
+            filep.write("    double %s;\n" % name)
+        elif isinstance(value, int):
+            filep.write("    int64_t %s;\n" % name)
+        else:
+            raise RuntimeError("unexpected type for: %s" % value)
+        if not omit_nl:
+            filep.write("\n")
+    for key in sorted(events.keys()):
+        filep.write("    /// Callbacks for '%s' event\n" % key)
+        filep.write("    std::vector<%sCallback> %s_cbs;\n" % (
+            event_name_to_camel_case(key), event_name_to_snake_case(key)))
+        filep.write("\n")
+        filep.write("    /// Allows to add callback for '%s' event using a fluent style\n" % key)
+        filep.write("    TaskInfo &on_%s(%sCallback &&cb) noexcept {\n" % (
+            event_name_to_snake_case(key), event_name_to_camel_case(key)))
+        filep.write("        %s_cbs.push_back(std::move(cb));\n" % event_name_to_snake_case(key))
+        filep.write("        return *this;\n")
+        filep.write("    }\n")
+        filep.write("\n")
+    for index, tpl in enumerate(sorted(sections["tasks"])):
+        name = tpl[0]
+        filep.write("    /// Factory method for creating a '%s' task\n" % (name))
+        filep.write("    static TaskInfo %s() noexcept {\n" % name)
+        filep.write("        TaskInfo info;\n")
+        filep.write("        info.name = \"%s\";\n" % name)
+        filep.write("        return info;\n")
+        filep.write("    };\n")
+        if index < len(sections["tasks"]) - 1:
+            filep.write("\n")
+    filep.write("};\n")
+    filep.write("\n")
+
+def write_nlohmann_json_exception(filep):
+    """ Writes the NlohmannJsonException class """
+    filep.write("/// Exceptions emitted by JSON library\n")
+    filep.write("#if NLOHMANN_JSON_VERSION_MAJOR >= 3\n")
+    filep.write("using NlohmannJsonException = nlohmann::json::exception;\n")
+    filep.write("#else\n")
+    filep.write("using NlohmannJsonException = std::exception;\n")
+    filep.write("#endif\n\n")
```

Comment:
Since `nlohmann::json::exception` derives from `std::exception`, this matter is actually much simpler than I initially anticipated, and it makes sense to update comments and see if we can update ASAP.